edgexfoundry/device-rest-go
# refactor(snap): edgex-snap-hooks related upgrade

This PR upgrades snap hooks to [2.3.0 ](https://github.com/canonical/edgex-snap-hooks/releases/tag/v2.3.0)with new function and refactor:
- Combine install hook and configure hook
- Use library function for copying directory
- Remove export bash command to process valid .env files
- Use Fatal for logging

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?) [Snap Testing device-rest](https://github.com/canonical/edgex-snap-testing/tree/main/test/suites/device-rest) passed with `FULL_CONFIG_TEST=true`
  <link to docs PR>

## Testing Instructions

1. Install and setup:
```
sudo snap install edgex-device-rest --channel=edge/pr-202
sudo snap set edgex-device-rest app-options=true
sudo snap set edgex-device-rest config.writable-loglevel=DEBUG
sudo snap set edgex-device-rest config.service-startupmsg="#test start up message#"
sudo snap start edgex-device-rest
```
2. Check the .env file is valid now without `export` command:
```
cat /var/snap/edgex-device-rest/current/config/device-rest/res/device-rest.env 

SERVICE_STARTUPMSG="#test start up message#"
WRITABLE_LOGLEVEL="DEBUG"
```
3. Validate env vars in `.env` have been exported successfully:
```
sudo snap logs -f edgex-device-rest

level=INFO ts=2022-06-14T14:06:02.834014222Z app=device-rest source=message.go:55 msg="#test start up message#"
level=DEBUG ts=2022-06-14T14:06:12.834954223Z app=device-rest source=executor.go:52 msg="xxx"
```
4. Validate all config files have been copied in right place:
```
tree /var/snap/edgex-device-rest/current/config
.
└── device-rest
    └── res
        ├── configuration.toml
        ├── device-rest.env
        ├── devices
        │   └── sample-devices.toml
        └── profiles
            ├── sample-image-device.yaml
            ├── sample-json-device.yaml
            └── sample-numeric-device.yaml

4 directories, 6 files

```

## New Dependency Instructions (If applicable)
